### DISCUSSION TITLE
.Net MAUI - Uno XAML Hot Reload on iOS Not working

### DISCUSSION THREAD

**Discussion by jpmock** [unoplatform/uno#15606] [Q&A]
Fresh install, everything checks out as installed and configured properly.
I am on a Mac and everything is latest and using .Net 8.
VSCode

I am just trying to test this out by changing some background colors on StackLayout and Labels and nothing works.  I made a color change, save the file, and nothing happens.  I have also restarted everything including the iOS Simulator.

I have checked the output window and under the Uno Platform - Hot Reload selection I can see:

[Error] Command failed: dotnet build /t:GetRemoteControlHostPath "/Users/me/MobileApps/myapp-mobile-maui/myapp.iOSCore/myapp.iOSCore.csproj" -r:iossimulator-x64 -p:UnoForceSingleTFM=true -f net8.0-ios
MSBuild version 17.9.4+90725d08d for .NET
  Determining projects to restore...
  Restored /Users/me/MobileApps/myapp-mobile-maui/myapp.ModelsCore/myapp.ModelsCore.csproj (in 445 ms).
  Restored /Users/me/MobileApps/myapp-mobile-maui/myapp.ServicesCore/myapp.ServicesCore.csproj (in 445 ms).
  Restored /Users/me/MobileApps/myapp-mobile-maui/myapp.iOSCore/myapp.iOSCore.csproj (in 503 ms).
  Restored /Users/me/MobileApps/myapp-mobile-maui/myapp.PortableCore/myapp.PortableCore.csproj (in 504 ms).
/Users/me/MobileApps/myapp-mobile-maui/myapp.iOSCore/myapp.iOSCore.csproj : error MSB4057: The target "GetRemoteControlHostPath" does not exist in the project. [TargetFramework=net8.0-ios]

Build FAILED.

/Users/me/MobileApps/myapp-mobile-maui/myapp.iOSCore/myapp.iOSCore.csproj : error MSB4057: The target "GetRemoteControlHostPath" does not exist in the project. [TargetFramework=net8.0-ios]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.13

-----------------------

I have gone through all the troubleshooting to no avail.

**Comment by sasakrsmanovic** [comment]
Hello. On what IDE are you; just to make everything is clear.

**Reply by jpmock** [reply]
Oops, that is a major detail I missed in the post, sorry.   I just updated it in my OP.  VSCode.

**Reply by sasakrsmanovic** [reply]
Thanks for updating. One of our engineers will take a look at this thread soon.

**Reply by jeromelaban** [reply]
Thanks for the report. Hot reload is not supported by the Uno Platform Extensions for MAUI apps, only debugging. Still, there should not be this specific error and we have an update coming soon.

**Reply by jpmock** [reply]
@jeromelaban I am not sure what you mean by not supported, only debugging?

I am trying to debug a MAUI app within VSCode and change XAML values, hit save, and see it update in the iOS Simulator.

Is that currently supported, or not yet?  The documentation seems to indicate it is supported and that only C# hot reload is NOT yet supported.

**Reply by jeromelaban** [reply]
Thanks, then you probably don't need the Uno Platform extension at this point, if you're doing only MAUI work. Why did you try the Uno Platform? Is it for the XAML intellisense?

**Reply by jpmock** [reply]
The only thing I need to get working is Hot Reload with XAML and by the looks of it, this is something the Uno Extension provides.  Am I wrong?  Please let me know.

Within VSCode I searched:  "XAML Hot Reload" and Uno came up in the results

From the description:  This extension is used with Uno Platform C#/XAML projects to develop and debug Uno Platform-built .NET Mobile, Web and Desktop Apps without leaving Visual Studio Code. **It also works for .NET MAUI mobile development and debugging.**

**Reply by jeromelaban** [reply]
Thanks. 

MAUI has its own extensions, which is likely what is at play for the hot reload of the MAUI app. Considering what you're mentioning, it is likely that if you uninstall the Uno extensions you'll get the exact same features as what you're using does not come from the Uno Platform extension, but the MAUI extension.
